perllib/Integrations/AccessPaySuite.pm

```diff
@@ -128,13 +127,28 @@ sub build_request_url {
 
 sub create_request {
     my ($self, $method, $url, $data) = @_;
-    my $content = ($method eq 'POST' || $method eq 'PUT') ? $self->build_form_data($data) : '';
 
-    return HTTP::Request->new(
-        $method => $url,
-        HTTP::Headers->new(%{ $self->headers }),
-        $content
-    );
+    my $headers = $self->headers();
+
+    if ($method eq 'POST') {
+        my $content = $self->build_form_data($data);
+        return HTTP::Request::Common::POST($url,
+            Content => $content,
+            %$headers
+        );
+    } elsif ($method eq 'PUT') {
+        my $content = $self->build_form_data($data);
+        return HTTP::Request::Common::PUT($url,
+            Content => $content,
+            %$headers
+        );
+    } elsif ($method eq 'GET') {
+        return HTTP::Request::Common::GET($url, %$headers);
+    } elsif ($method eq 'DELETE') {
+        return HTTP::Request::Common::DELETE($url, %$headers);
+    } else {
+        die "Unsupported method: $method";
+    }
```

Comment:
I think this can be reduced to:
```suggestion
    return HTTP::Request::Common->can($method)->($url,
        $method eq 'POST' || $method eq 'PUT' ? $data : (),
        %$headers
    );
```
(if you supply $data as the second argument for post/put it will build the form data for you)